Map/Source/SquareMap.cpp

```diff
@@ -62,4 +63,133 @@ SquareMap::SquareMap(const int mapSize)
     {
         _fieldList.at(i).at(0) = std::make_unique<FightField>();
     }
-}
\ No newline at end of file
+}
+
+SquareMap::SquareMap(FieldList&& fieldList) : _fieldList(std::move(fieldList))
+{
+}
+
+bool SquareMap::isFieldAccessible(const Position& position)
+{
+    if (not isField(position) or getField(position)->getType() == FieldType::Wall)
+    {
+        return false;
+    }
+    return true;
+}
+
+bool SquareMap::isField(const Position& position)
+{
+    try
+    {
+        if (getField(position))
+        {
+            return true;
+        }
+        return false;
+    }
+    catch (std::out_of_range&)
+    {
+        return false;
+    }
+}
+
+const std::unique_ptr<Field>& SquareMap::getField(const Position& position)
+{
+    return _fieldList.at(position._x).at(position._y);
+}
+
+void SquareMap::updateVisilibity(const Position& playerPosition)
+{
+    makeNonBarrierFieldsInvisible();
+    if (getField(playerPosition)->getType() == FieldType::Door)
+    {
+        makeRoomVisible(Position(playerPosition._x + 1, playerPosition._y));
+        makeRoomVisible(Position(playerPosition._x - 1, playerPosition._y));
+        makeRoomVisible(Position(playerPosition._x, playerPosition._y + 1));
+        makeRoomVisible(Position(playerPosition._x, playerPosition._y - 1));
+    }
+    else if(not isFieldAccessible(playerPosition))
+    {
+        return;
+    }
+    else
+    {
+        makeRoomVisible(playerPosition);
+    }
+}
+
+void SquareMap::makeNonBarrierFieldsInvisible()
```

Comment:
To jest zbędne i damiarowe. Dużo prosciej jest ustawić te wartości na true/false w konstruktorach pol, niz wyciagac ta logike do klasy mapa. Było by deafult że pola sa isVisible = false, i wrócmy do tego stanu (zmieniles to zachowanie). Wtedy można wywalić tą metodę

Reply:
Zresztą widze ze juz wróciłeś do starego zachowania. W takim wypadku ta metoda kompletnie nic nie wnosi. Można ją usunąc razem z Field::makeInvisible()